Automattic/pocket-casts-ios
# Feature: Episode rich text description

| 📘 Part of: #2479  |  
|:---:|

Fixes # 


This PR adds support for rich html description for podcasts

| 1 | 2 | 3 | 4 |
| - | - | - | - |
| ![Simulator Screenshot - iPhone 16 Pro - 2024-12-11 at 10 46 49](https://github.com/user-attachments/assets/0db304f3-2661-4e7d-b7f9-6f985ebd5166) | ![Simulator Screenshot - iPhone 16 Pro - 2024-12-11 at 10 46 31](https://github.com/user-attachments/assets/1aabdabb-7bc2-43af-9986-79ff981c93c3) | ![Simulator Screenshot - iPhone 16 Pro - 2024-12-11 at 10 18 24](https://github.com/user-attachments/assets/4fc4ecb1-809e-4d58-8fe8-735be421e229) |  ![Simulator Screenshot - iPhone 16 Pro - 2024-12-11 at 10 16 59](https://github.com/user-attachments/assets/1f9f3739-2f97-493f-a864-8c76a59b2b76) |


## To test

1. Start the app using the **staging** environment.
2. Ensure that you have the `usePodcastHTMLDescription` FF enabled.
3. Open the Discovery tab
4. Open a podcast with rich html descriptions. Ex: DirtBag Rich
5. Check if the description shows correctly. This is: links are show with underline, italic show in the last paragraph.
6. Detect if you can tap on the description links and it will open Safari with the URL
7. Tap on the description to expand 
8. Tap on the top arrow to collapse the description
9. Tap on the top arrow to expand the description
10. Change tab
11. Go back to the discovery tab
12. Check if description is showing correctly
13. Another podcast to check is The Daily, this description does not have html but we are replacing the `\n` for `<br>` 
14. Open the podcast description and check if the different paragraphs have a empty line separating them.

## Checklist
